File: app/src/main/java/uk/co/ourfriendirony/medianotifier/general/IntentGenerator.java
package uk.co.ourfriendirony.medianotifier.general;

import android.content.Intent;
import android.net.Uri;

import androidx.annotation.NonNull;

public class IntentGenerator {
    @NonNull
    public static Intent getWebPageIntent(String webAddress) {
        Intent intent = new Intent();
        intent.setAction(Intent.ACTION_VIEW);
        intent.addCategory(Intent.CATEGORY_BROWSABLE);
        intent.setData(Uri.parse(webAddress));
        return intent;
    }

    public static Intent getContactEmailIntent() {
        String emailToAddress = "ourfriendirony@gmail.com";
        String emailSubject = "[MediaNotifier] I've got this to say about your app...";
        Intent emailIntent = new Intent(android.content.Intent.ACTION_SEND);

        emailIntent.setType("plain/text");
        emailIntent.putExtra(android.content.Intent.EXTRA_EMAIL, new String[]{emailToAddress});
        emailIntent.putExtra(android.content.Intent.EXTRA_SUBJECT, emailSubject);

        return emailIntent;
    }
}

File: app/src/main/kotlin/uk/co/ourfriendirony/medianotifier/general/IntentGenerator.kt
package uk.co.ourfriendirony.medianotifier.general

import android.content.Intent
import android.net.Uri

object IntentGenerator {
    @JvmStatic
    fun getWebPageIntent(webAddress: String?): Intent {
        val intent = Intent()
        intent.action = Intent.ACTION_VIEW
        intent.addCategory(Intent.CATEGORY_BROWSABLE)
        intent.data = Uri.parse(webAddress)
        return intent
    }

    @JvmStatic
    val contactEmailIntent: Intent
        get() {
            val emailToAddress = "ourfriendirony@gmail.com"
            val emailSubject = "[MediaNotifier] I've got this to say about your app..."
            val emailIntent = Intent(Intent.ACTION_SEND)
            emailIntent.type = "plain/text"
            emailIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf(emailToAddress))
            emailIntent.putExtra(Intent.EXTRA_SUBJECT, emailSubject)
            return emailIntent
        }
}